sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client.hpp

```diff
@@ -0,0 +1,97 @@
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+//
+// Code generated by Microsoft (R) AutoRest C++ Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/context.hpp>
+#include <azure/core/internal/http/pipeline.hpp>
+#include <azure/core/nullable.hpp>
+#include <azure/core/response.hpp>
+#include <azure/core/url.hpp>
+#include <azure/keyvault/administration/dll_import_export.hpp>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Administration {
+  /**
+   * @brief The type specifier of the value.
+   */
+  class SettingTypeEnum final {
+  public:
+    /**
+     * @brief Default constructor
+     */
+    SettingTypeEnum() = default;
+
+    /**
+     * @brief Constructor
+     * @param value string value
```

Comment:
I suspect the value is something more than just a string.  Are there values that are acceptable or not acceptable.